libraries/adaptive-expressions/src/functionUtils.internal.ts

```diff
@@ -84,10 +83,15 @@ export class InternalFunctionUtils {
                         if (!error) {
                             propertyName = propertyName || '';
                         }
+
+                        const sortBy = (key: string) => {
+                            return (a: unknown, b: unknown) => (a[key] > b[key] ? 1 : b[key] > a[key] ? -1 : 0);
+                        };
+
                         if (isDescending) {
-                            result = sortBy(arr, propertyName).reverse();
+                            result = arr.concat().sort(sortBy(propertyName)).reverse();
```

Comment:
I imagine calling `concat()` ensures we don't mutate `arr` in place - leaving a comment explaining that would be useful since it isn't immediately obvious.

Reply:
Yes. But after I double checked the code, this part could be removed. Because `const arr: any = oriArr.slice(0);` this line has done it.